az-hadoop-jobtype-plugin/src/main/java/azkaban/jobtype/IHadoopJob.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2019 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package azkaban.jobtype;
+
+public interface IHadoopJob {
```

Comment:
i'm not sure why we need this interface. Could you elaborate?

Reply:
Well. The way to setup HadoopJobProperties by the HadoopProxy setting is case by case. For instance, HadoopSpark will be different to HadoopShell. We use this way to standardize the implementation.